### PR TITLE
Add scraper for logs

### PR DESCRIPTION
**Pull request by sincejune**
#### Description
Add scraper for logs

#### Link to tracking issue
Relates to #11238 

#### Testing
Added unit tests

#### Documentation
Added

### CODE REVIEW

File: scraper/logs.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/plog"
+)
+
+// Logs is the base interface for logs scrapers.
+type Logs interface {
+	component.Component
+
+	ScrapeLogs(context.Context) (plog.Logs, error)
```
Comment (atoulme) [inline scraper/logs.go:17]: consider adding a comment to this function

**Review by atoulme — APPROVED** [review]

**Review by bogdandrutu — COMMENTED** [review]
You also need to add this to the factory.

File: scraper/logs.go (hunk 1 above)
Comment (sincejune) [reply scraper/logs.go:18]: Updated

**Comment by sincejune** [comment]
> You also need to add this to the factory.

I'll add this change in the next PR.

**Comment by sincejune** [comment]
> > You also need to add this to the factory.
> 
> I'll add this change in the next PR.

Added in #11822 cc @bogdandrutu

**Pull request merged**